processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -160,13 +160,24 @@ public void receive(ConsumerRecord<byte[], byte[]> record) {
         setName(String.format("DecatonSubscriptionThread-%s", scope));
     }
 
+
+    public ProcessorSubscription(SubscriptionScope scope,
```

Comment:
We don't need another constructor and pulling `SubPartitionerSupplier` around multiple classes.

We can just add new field in `SubscriptionScope`.

Then, in `PartitionProcessor`'s constructor, it can be accessible through `PartitionScope` constructor parameter because `PartitionScope` extends `SubscriptionScope`